module/zfs/spa.c

```diff
@@ -4650,6 +4805,145 @@ spa_ld_check_features(spa_t *spa, boolean_t *missing_feat_writep)
 	return (0);
 }
 
+struct load_chain_map_arg {
+	blkptr_t *bp;
+	spa_t *spa;
+	uint64_t *error;
+};
+
+static int
+load_chain_map_claim_blk_cb(spa_t *spa, const blkptr_t *bp, void *arg)
+{
+	(void) arg;
+	int error = metaslab_claim(spa, bp,
+	    spa_get_dsl(spa)->dp_tx.tx_open_txg);
```

Comment:
On a level of suspicion: you are claiming blocks in whatever the current open txg is, while as I see normal ZIL uses `dmu_tx_create_assigned(dp, spa_first_txg(spa))` result for all the claiming process.

Reply:
I believe the two should be the same, but I can certainly change it to spa_first_txg for consistency